Tools/clinic/clinic.py

```diff
@@ -4602,7 +4609,10 @@ def parse(self, block: Block) -> None:
         for line_number, line in enumerate(lines, self.clinic.block_parser.block_start_line_number):
             if '\t' in line:
                 fail('Tab characters are illegal in the Clinic DSL.\n\t' + repr(line), line_number=block_start)
-            self.state(line)
+            try:
+                self.state(line)
+            except ClinicError as exc:
+                raise ClinicError(str(exc), filename=exc.filename, lineno=line_number)
```

Comment:
IMO, this is a workaround! We should ensure that line numbers are correctly updated in the state machine. Let's create an issue about this.

Reply:
I'd like to keep the workaround for now; it really helps to have accurate line numbers when debugging.